geode-core/src/main/java/org/apache/geode/internal/cache/GemFireCacheImpl.java

```diff
@@ -2997,12 +2995,12 @@ public void validatePoolFactory(PoolFactory poolFactory) {
           } finally {
             // clean up if initialize fails for any reason
             setRegionByPath(region.getFullPath(), null);
-            synchronized (rootRegions) {
-              Region rootRegion = rootRegions.get(name);
-              if (rootRegion == region) {
-                rootRegions.remove(name);
-              }
+
+            Region rootRegion = rootRegions.get(name);
```

Comment:
This does not look thread safe. Instead of calling get and then remove you need a single call.
Here I think you want: rootRegions.remove(name, region);